core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -591,6 +659,13 @@ object Infer {
           } yield TypedExpr.coerceRho(t1, ck) // TODO this coerce seems right, since we have unified
       })
 
+    def setInf(inf: Expected.Inf[(Type.Rho, Region)], rho: Type.Rho, r: Region): Infer[Unit] =
+      lift(inf.ref.update[Infer[Unit]] {
+        case Left(_) => (Right((rho, r)), unit)
+        case right@Right((rho1, r1)) => (right, unify(rho1, rho, r1, r))
```

Comment:
Seems we never hit this branch so the code only solves one or zero times for each variable which is what we would hope.

Probably revert this change.